test/parallel/test-process-env-allowed-flags.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const assert = require('assert');
+require('../common');
+
+const goodFlags = [
+  '--inspect-brk',
+  'inspect-brk',
+  '--perf_basic_prof',
+  '--perf-basic-prof',
+  'perf-basic-prof',
+  '--perf_basic-prof',
+  'perf_basic-prof',
+  'perf_basic_prof',
+  '-r',
+  'r',
+  '--stack-trace-limit=100',
+  '--stack-trace-limit=-=xX_nodejs_Xx=-'
+];
+
+const badFlags = [
+  '--inspect_brk',
+  'INSPECT-BRK',
+  '--INSPECT-BRK',
+  '--r',
+  '-R',
+  '---inspect-brk',
+  '--cheeseburgers'
+];
+
+goodFlags.forEach((flag) => {
+  assert.strictEqual(
+    process.allowedNodeEnvironmentFlags.has(flag),
+    true,
+    `flag should be in set: ${flag}`
+  );
+});
+
+badFlags.forEach((flag) => {
+  assert.strictEqual(
+    process.allowedNodeEnvironmentFlags.has(flag),
+    false,
+    `flag should not be in set: ${flag}`
+  );
+});
+
+process.allowedNodeEnvironmentFlags.forEach((flag) => {
+  assert.strictEqual(/^--?[a-z8_-]+$/.test(flag), true);
+});
+
+assert.strictEqual(Object.isFrozen(process.allowedNodeEnvironmentFlags), true);
+
+process.allowedNodeEnvironmentFlags.add('foo');
+assert.strictEqual(process.allowedNodeEnvironmentFlags.has('foo'), false);
+process.allowedNodeEnvironmentFlags.forEach((flag) => {
+  assert.strictEqual(flag === 'foo', false);
+});
+
+process.allowedNodeEnvironmentFlags.clear();
```

Comment:
I've been getting in the habit of putting a comment above each test case lexically scoping the test, so that:

1. it's less likely that test setup and teardown get clumped together.
2. you have a description of what's being tested for.

```
// should return false if flag is not in allowedNodeEnvironmentFlags
{
  process.allowedNodeEnvironmentFlags.add('foo');
  assert.strictEqual(process.allowedNodeEnvironmentFlags.has('foo'), false);
  process.allowedNodeEnvironmentFlags.forEach((flag) => {
    assert.strictEqual(flag === 'foo', false);
  });
}
```

CC: @ratracegrad

Reply:
I've done some grouping in this manner